cdap-app-fabric/src/main/java/io/cdap/cdap/security/auth/AuditLogSingleTopicSubscriberService.java

```diff
@@ -130,29 +126,26 @@ protected void storeMessageId(StructuredTableContext context, String messageId)
    */
   @Override
   protected void processMessages(StructuredTableContext structuredTableContext,
-                                 Iterator<ImmutablePair<String, AuditLogContext>> messages) throws Exception {
+                                 Iterator<ImmutablePair<String, AuditLogRequest>> messages) throws Exception {
 
-    Queue<AuditLogContext> auditLogContextQueue = new ArrayDeque<>();
+    List<AuditLogRequest> auditLogRequestList = new ArrayList<>();
 
     while (messages.hasNext()) {
-      ImmutablePair<String, AuditLogContext> next = messages.next();
-      AuditLogContext auditLogContext = next.getSecond();
-      if (auditLogContext.isAuditLoggingRequired()){
-        auditLogContextQueue.add(auditLogContext);
-      }
+      ImmutablePair<String, AuditLogRequest> next = messages.next();
+      auditLogRequestList.add(next.getSecond());
     }
 
-    if (!auditLogContextQueue.isEmpty()) {
-      LOG.debug("Publishing a queue of Audit Log events of size {} events.", auditLogContextQueue.size());
+    if (!auditLogRequestList.isEmpty()) {
+      LOG.debug("Publishing a queue of Audit Log events of size {} events.", auditLogRequestList.size());
       AuditLoggerSpi.PublishStatus publishStatus =
-        this.accessControllerInstantiator.get().publishAuditLogs(auditLogContextQueue);
+        this.accessControllerInstantiator.get().publishAuditLogs(auditLogRequestList);
       // TODO : This logic can change based on how Auth Ext publishes a batch.
-      if (publishStatus.equals(AuditLoggerSpi.PublishStatus.UNSUCCESSFUL)) {
+      if (publishStatus == null || publishStatus.equals(AuditLoggerSpi.PublishStatus.UNSUCCESSFUL)) {
```

Comment:
It was already here, but you don't need equals for enums. I would just do `publishStatus != PUBLISHED`